Makefile

```diff
@@ -12,72 +12,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-SHELL = /bin/bash
-
-GO_VERSION = 1.12.5
+GO_VERSION = 1.13.8
 
 export CGO_ENABLED := 0
 
-export E2E_SSH_PUBKEY ?= $(shell cat ~/.ssh/id_rsa.pub)
+export E2E_SSH_PUBKEY ?= $(shell test -f ~/.ssh/id_rsa.pub && cat ~/.ssh/id_rsa.pub)
 
-export GIT_TAG ?= $(shell git tag --points-at HEAD)
+export DOCKER_TAG ?= $(shell git tag --points-at HEAD)
 
 REGISTRY ?= docker.io
 REGISTRY_NAMESPACE ?= kubermatic
 
+LDFLAGS ?= -ldflags '-s -w'
+
 IMAGE_TAG = \
 		$(shell echo $$(git rev-parse HEAD && if [[ -n $$(git status --porcelain) ]]; then echo '-dirty'; fi)|tr -d ' ')
 IMAGE_NAME = $(REGISTRY)/$(REGISTRY_NAMESPACE)/machine-controller:$(IMAGE_TAG)
 
+OS = centos coreos ubuntu sles rhel
+USERDATA_BIN = $(patsubst %, machine-controller-userdata-%, $(OS))
+
+.PHONY: all
+all: machine-controller webhook $(USERDATA_BIN)
+
+machine-controller-userdata-%: cmd/userdata/% $(shell find cmd/userdata/$* pkg -name '*.go')
+	go build -v \
+		$(LDFLAGS) \
+		-o $@ \
+		github.com/kubermatic/machine-controller/cmd/userdata/$*
+
+%: cmd/% $(shell find cmd/$* pkg -name '*.go')
+	go build -v \
+		$(LDFLAGS) \
+		-o $@ \
+		github.com/kubermatic/machine-controller/cmd/$*
 
-vendor: Gopkg.lock Gopkg.toml
-	dep ensure -vendor-only
+.PHONY: clean
+clean:
+	rm -f machine-controller \
+		webhook \
+		$(USERDATA_BIN)
 
+.PHONY: machine-controller-docker
 machine-controller-docker:
 	@docker run --rm \
 		-v $$PWD:/go/src/github.com/kubermatic/machine-controller \
 		-v $$PWD/.buildcache:/cache \
 		-e GOCACHE=/cache \
 		-w /go/src/github.com/kubermatic/machine-controller \
 		golang:$(GO_VERSION) \
-			make machine-controller
-
-.PHONY: machine-controller
```

Comment:
I think after removing this, the `run-machine-controller.sh` in the hack folder will fail or?

Reply:
Good catch, `machine-controller` target is now covered by [this](https://v1-15.docs.kubernetes.io/docs/setup/release/notes/#unchanged) pattern rule .The problem is that it does not include the build of the userdata binaries that are required for the script to run. I have either to use `all` target or a newly introduced target that includes only `machine-controller` and userdata.